src/main/java/org/apache/commons/io/build/AbstractOrigin.java

```diff
@@ -402,6 +402,13 @@ public Path getPath() {
             return Paths.get(get());
         }
 
+        @Override
+        public InputStream getInputStream(final OpenOption... options) throws IOException {
+            if ("http".equalsIgnoreCase(get().getScheme()) || "https".equalsIgnoreCase(get().getScheme())) {
```

Comment:
This method should only call `get()` and `getScheme()` once.

Don't override the file systems for these two schemes. You should check if a file system is installed already that handles HTTP[S] before overriding NIO Path handling; use `org.apache.commons.io.file.spi.FileSystemProviders`.

Reply:
Hope that I understand you correctly, please help review it 🙇